jax/numpy/lax_numpy.py

```diff
@@ -92,7 +93,11 @@
 nan = np.nan
 
 # And some numpy utility functions
-set_printoptions = np.set_printoptions
+def printoptions(pytree,floatmode="unique"):
```

Comment:
This doesn't seem like it's doing something sensible to me. The code you are copying from in #3438 is printing a tree of arrays by first converting arrays to NumPy and then relying on NumPy to do the right thing with respect to precision. However that's not what you want here; instead we want to change how `DeviceArray` values print themselves. I don't see how the change you have made here could work.

I suspect you want to do is to change the implementation of `DeviceArray.__repr__` to take more of the NumPy options into account. Its implementation is here:
https://github.com/google/jax/blob/a6e3f992a70775ef16c285e024cd2ac62ea32077/jax/interpreters/xla.py#L1051

I suggest reverting changes to this file and to try modifying that function.

Reply:
Thank you for the suggestion and for indicating the location of the implementation. You are right, we should change how `DeviceArrays` are printed instead of using such workaround. It turns out that because there is an implementation of `DeviceArray.__str__`, the function `DeviceArray.__repr__` is [ignored](https://stackoverflow.com/questions/1436703/difference-between-str-and-repr). So I suppose the implementation of `__str__` should be modified. Now I am trying to figure you where and why the `floatmode="unique"` option is apparently ignored.